grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1560,38 +1662,59 @@ class StockMovementService {
      * @param stockMovementItem
      * @return
      */
-    List getPickPageItems(StockMovementItem stockMovementItem) {
+    List getPickPageItems(StockMovementItem stockMovementItem, Boolean createNewPicklistItems = true) {
```

Comment:
I don't feel good about chaning this method in this way. 

Can you explain what this is doing and why it's necessary?
Is it possible that we could leave getPickPageItems() alone since it's being used elsewhere, but then transform those into whatever you're trying to do here. 
```
List pickPageItems = getPickPageItems(stockMovementItem)
List pickPageItemsUltra = pickPageItems.collect { transformToSomethingUltra(it) }
```

Also, I would like to find another name for the createNewPicklistItems argument. It's possible I just don't understand what's happening here and it will become more intuitive once I understand the reason.

Reply:
The main issue here was that `getPickPageItems` creates new picklist for items that don't have the picklist.

What I was trying to achieve is to specify to the method that I don't want to create a new picklist for items that don't have it, all I want is to get the existing data.